packages/notebook/src/notebookfooter.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) Jupyter Development Team.
+ * Distributed under the terms of the Modified BSD License.
+ */
+
+import { Widget } from '@lumino/widgets';
+import { Message } from '@lumino/messaging';
+import { Notebook } from './widget';
+import { NotebookActions } from './actions';
+
+const NOTEBOOK_FOOTER_CLASS = 'jp-Notebook-footer';
+
+/**
+ * A footer widget added after the last cell of the notebook.
+ */
+export class NotebookFooter extends Widget {
+  /**
+   * Construct a footer widget.
+   */
+  constructor(protected notebook: Notebook) {
+    super();
+    this.addClass(NOTEBOOK_FOOTER_CLASS);
+    this._createHiddenButton();
+    const text = document.createElement('p');
+    text.innerText = 'Click to add a cell.';
+    this.node.appendChild(text);
+  }
+
+  /**
+   * Handle incoming events.
+   */
+  handleEvent(event: Event): void {
+    switch (event.type) {
+      case 'click':
+        this.onClick(event as MouseEvent);
+        break;
+      case 'keydown':
+        if ((event as KeyboardEvent).key === 'ArrowUp') {
+          this.onArrowUp();
+          break;
+        }
+    }
+  }
+
+  /**
+   * On single click (mouse event), insert a cell below (at the end of the notebook as default behavior).
+   */
+  onClick(event: any): void {
+    NotebookActions.insertBelow(this.notebook);
+  }
```

Comment:
Please make this protected and not public.
Don't request the event if you don't need it.
You should set the active index to the latest cell otherwise the cell does not get inserted where it should.

```suggestion
  protected onClick(): void {
    if (this.notebook.cells.length > 0) {
      this.notebook.activeCellIndex = this.notebook.cells.length - 1;
    }
    NotebookActions.insertBelow(this.notebook);
  }
```